graspologic/pipeline/embed/omnibus_embedding.py

```diff
@@ -0,0 +1,280 @@
+          For directed graphs, if ``elbow_cut==None``, you will receive an embedding that
+          has ``nodes`` rows and ``2*dimensions`` columns.
+          If ``elbow_cut`` is specified to be not ``None``, we will cut the embedding at
+          ``elbow_cut`` elbow, but the provided ``dimensions`` will be used in the
+          creation of the SVD.
+      elbow_cut : Optional[int] (default=None)
+          Using a process described by Zhu & Ghodsi in their paper "Automatic
+          dimensionality selection from the scree plot via the use of profile likelihood",
+          truncate the dimensionality of the return on the ``elbow_cut``-th elbow.
+          By default this value is ``None`` but can be used to reduce the dimensionality
+          of the returned tensors.
+      svd_solver_algorithm : str (default="randomized")
+          allowed values: {'randomized', 'full', 'truncated'}
+
+          SVD solver to use:
+
+              - 'randomized'
+                  Computes randomized svd using
+                  :func:`sklearn.utils.extmath.randomized_svd`
+              - 'full'
+                  Computes full svd using :func:`scipy.linalg.svd`
+                  Does not support ``graph`` input of type scipy.sparse.csr_matrix
+              - 'truncated'
+                  Computes truncated svd using :func:`scipy.sparse.linalg.svds`
+      svd_solver_iterations : int (default=5)
+          Number of iterations for randomized SVD solver. Not used by 'full' or
+          'truncated'. The default is larger than the default in randomized_svd
+          to handle sparse matrices that may have large slowly decaying spectrum.
+      svd_seed : Optional[int] (default=None)
+          Used to seed the PRNG used in the ``randomized`` svd solver algorithm.
+      weight_attribute : str (default="weight")
+          The edge dictionary key that contains the weight of the edge.
+
+      Returns
+      -------
+      List[Tuple[Embeddings, Embeddings]]
+
+      Raises
+      ------
+      beartype.roar.BeartypeCallHintPepParamException if parameters do not match type hints
+      ValueError if values are not within appropriate ranges or allowed values
+
+      See Also
+      --------
+      graspologic.pipeline.embed.Embeddings
+      graspologic.embed.OmnibusEmbed
+      graspologic.embed.AdjacencySpectralEmbed
+      graspologic.embed.select_svd
+
+      References
+      ----------
+      .. [1] Sussman, D.L., Tang, M., Fishkind, D.E., Priebe, C.E.  "A
+         Consistent Adjacency Spectral Embedding for Stochastic Blockmodel Graphs,"
+         Journal of the American Statistical Association, Vol. 107(499), 2012
+
+      .. [2] Levin, K., Roosta-Khorasani, F., Mahoney, M. W., & Priebe, C. E. (2018).
+          Out-of-sample extension of graph adjacency spectral embedding. PMLR: Proceedings
+          of Machine Learning Research, 80, 2975-2984.
+
+      .. [3] Zhu, M. and Ghodsi, A. (2006). Automatic dimensionality selection from the

```

Comment:
we should absolutely reference the paper from https://github.com/microsoft/graspologic/blob/dev/graspologic/embed/omni.py#L132